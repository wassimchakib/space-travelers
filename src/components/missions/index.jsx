import './styles.css';

const Missions = () => (
  <section>
    <div className="container">
      <table className="table-container">
        <thead id="thead">
          <tr>
            <th className="th">Missions</th>
            <th className="th">Description</th>
            <th className="th">status</th>
            <th className="th" aria-label="empty-table-head" />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td className="td mission-title">Thaicom</td>
            <td className="td">
              Lorem ipsum dolor, sit amet consectetur adipisicing elit. Molestiae esse consectetur
              inventore quod possimus veniam ipsum ab quam recusandae ad eius minus in nulla
              veritatis, dolorem doloremque harum sed non, itaque nesciunt? Dolorum dolore
              asperiores debitis! Exercitationem, optio esse quasi nisi in, obcaecati fuga eum
              cumque mollitia aliquid culpa officia.
            </td>
            <td className="td">
              <span className="status-not-a-member">NOT A MEMBER</span>
            </td>
            <td className="td">
              <button className="join-mission-btn" type="button">
                Join Mission
              </button>
            </td>
          </tr>
          <tr>
            <td className="td">Thaicom</td>
            <td className="td">
              Lorem ipsum dolor, sit amet consectetur adipisicing elit. Molestiae esse consectetur
              inventore quod possimus veniam ipsum ab quam recusandae ad eius minus in nulla
              veritatis, dolorem doloremque harum sed non, itaque nesciunt? Dolorum dolore
              asperiores debitis! Exercitationem, optio esse quasi nisi in, obcaecati fuga eum
              cumque mollitia aliquid culpa officia.
            </td>
            <td className="td">
              <span className="status-not-a-member">NOT A MEMBER</span>
            </td>
            <td className="td">
              <button className="join-mission-btn" type="button">
                Join Mission
              </button>
            </td>
          </tr>
          <tr>
            <td className="td">Thaicom</td>
            <td className="td">
              Lorem ipsum dolor, sit amet consectetur adipisicing elit. Molestiae esse consectetur
              inventore quod possimus veniam ipsum ab quam recusandae ad eius minus in nulla
              veritatis, dolorem doloremque harum sed non, itaque nesciunt? Dolorum dolore
              asperiores debitis! Exercitationem, optio esse quasi nisi in, obcaecati fuga eum
              cumque mollitia aliquid culpa officia.
            </td>
            <td className="td">
              <span className="status-not-a-member">NOT A MEMBER</span>
            </td>
            <td className="td">
              <button className="join-mission-btn" type="button">
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
);

export default Missions;
