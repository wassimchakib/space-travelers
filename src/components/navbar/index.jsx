import { NavLink } from 'react-router-dom';
import logo from '../../assets/planet.png';
import './navbar.css';

const Navbar = () => (
  <div>
    <img src={logo} alt="planet-logo" />
    <h1>Space Travelers</h1>
    <ul>
      <li>
        <NavLink to="rockets" activeclassname="active">
          Rockets
        </NavLink>
      </li>
      <li>
        <NavLink to="missions" activeclassname="active">
          Missions
        </NavLink>
      </li>
      <li>
        <NavLink to="profile" activeclassname="active">
          Profile
        </NavLink>
      </li>
    </ul>
  </div>
);

export default Navbar;
