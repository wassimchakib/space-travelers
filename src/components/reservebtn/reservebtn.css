.rocket__btn {
  font-family: 'Poppins', sans-serif;
  padding: var(--fs-1);
  border: none;
  border-radius: 5px;
  cursor: pointer;
}

.rocket__reserve {
  background-color: #007bff;
  color: #fff;
}

.rocket__reserve:hover {
  background-color: #4ca1fb;
}

.rocket__cancel {
  background-color: #fff;
  color: #d5d3d3;
  border: 1px solid #e8e8e8;
}

.rocket__cancel:hover {
  background-color: #f1f1f1;
}
