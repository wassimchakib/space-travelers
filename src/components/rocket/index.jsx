import PropTypes from 'prop-types';
import ReserveBtn from '../reservebtn';
import './rocket.css';

const Rocket = ({ rocket }) => {
  const {
    id, images, name, description,reserved
  } = rocket;

  return (
    <div className="rocket">
      <img className="rocket__image" src={images[Math.floor(Math.random() * images.length)]} alt={name} />
      <div className="rocket__info">
        <h2>{name}</h2>
        <p>
          {reserved && <span className="rocket__reserved">Reserved</span>}
          {description}</p>
        <div>
          <ReserveBtn reserved={reserved} id={id} />
        </div>
      </div>
    </div>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string),
    reserved: PropTypes.bool.isRequired,
  }).isRequired,
};

export default Rocket;
