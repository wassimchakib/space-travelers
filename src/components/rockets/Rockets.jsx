import './rockets.css';

const Rockets = () => (
  <section className="rocket">
    <div className="container">
      <table>
        <thead>
          <tr>
            <td>Mission</td>
            <td>Description</td>
            <td>Status</td>
            <td />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Thaicom</td>
            <td>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Excepturi
              ducimus, dolore sequi et impedit molestiae odit quam repellendus
              quis dignissimos delectus vitae libero saepe similique quia eos.
              Incidunt, provident molestias.
            </td>
            <td><span className="tag">Not A Member</span></td>
            <td><button type="button">Join Mission</button></td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
);

export default Rockets;
