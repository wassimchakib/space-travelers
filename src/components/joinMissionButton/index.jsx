import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { joinMission } from '../../redux/missions/missionsSlice';
import './joinMissionButton.css';

const JoinMissionButton = ({ joinedMission, missionId }) => {
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(joinMission(missionId));
  };
  const missionStatusClass = joinedMission ? 'leave-missioin-btn' : 'join-mission-btn';

  return (
    <button className={`${missionStatusClass}`} type="button" onClick={handleClick}>
      {joinedMission ? 'Leave Mission' : 'Join Mission'}
    </button>
  );
};

JoinMissionButton.propTypes = {
  joinedMission: PropTypes.bool.isRequired,
  missionId: PropTypes.string.isRequired,
};

export default JoinMissionButton;
