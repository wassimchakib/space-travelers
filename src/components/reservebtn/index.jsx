import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { reserveRocket } from '../../redux/rockets/rocketsSlice';
import './reservebtn.css';

const ReserveBtn = ({ id, reserved }) => {
  const dispatch = useDispatch();

  const handleClick = (id) => {
    dispatch(reserveRocket(id));
  }
  return (
    <button
      className={ `rocket__btn ${reserved ? "rocket__cancel" : "rocket__reserve"}`}
      type="button"
      onClick={() => handleClick(id)}
    >
      {reserved ? "Cancel Reservation" : "Reserve Rocket" }
    </button>
  );
};

ReserveBtn.propTypes = {
  id: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default ReserveBtn;
