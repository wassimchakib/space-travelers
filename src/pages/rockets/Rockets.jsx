import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Rocket from '../../components/rocket';
import { getRockets } from '../../redux/rockets/rocketsSlice';
import './rockets.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets);

  useEffect(() => {
    if (rockets.rockets.length === 0) dispatch(getRockets());
  }, [dispatch, rockets.rockets.length]);
  return (
    <section className="rocket">
      <div className="container">
        {rockets.loading && <h1>It is loading</h1>}
        {!rockets.loading && rockets.rockets.map((rocket) => (
          <Rocket key={rocket.id} rocket={rocket} />
        ))}
      </div>
    </section>
  );
};

export default Rockets;
