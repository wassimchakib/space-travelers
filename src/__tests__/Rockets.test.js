import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import * as btn from '../components/reservebtn';
import ReserveBtn from '../components/reservebtn';
import Rocket from '../components/rocket';
import Rockets from '../pages/rockets/Rockets'

jest.mock('react-redux');

describe('Rocket page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Render rocket component', async () => {
    jest.spyOn(btn, 'default').mockImplementation();
    const obj = {
      id: '1',
      images: [],
      name: 'Falcon 1',
      description: 'Falcon 1 description',
      reserved: false,
    };
    const tree = render(<Rocket rocket={obj} />);
    const title = tree.getByText('Falcon 1');
    expect(tree).toMatchSnapshot();
    expect(title.textContent).toBe('Falcon 1');
  });

  it('renders rockets page', () => {
    useSelector.mockReturnValue({
      loading: false,
      error: '',
      rockets: [
        {
          id: "2",
          name: "Falcon 3",
          description: "Falcon description",
          images: [],
          reserved: false,
          wikipedia: "Https://www.wikipedia.com/falcon1",
        }
      ]
    }
    );

    const tree = render(<Rockets />)

    expect(tree).toMatchSnapshot();
  });

  it('Should render Cancel Reservation button when reserved = true', () => {
    const tree = render(<ReserveBtn id="falcon1" reserved />)
    expect(tree).toMatchSnapshot();
  });

  it('Should render Reserve Rocket button when reserved = true', () => {
    const tree = render(<ReserveBtn id="falcon1" reserved={false} />)
    expect(tree).toMatchSnapshot();
  });
});
