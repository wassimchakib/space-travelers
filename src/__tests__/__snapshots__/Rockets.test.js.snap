// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket page Render rocket component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocket"
      >
        <img
          alt="Falcon 1"
          class="rocket__image"
        />
        <div
          class="rocket__info"
        >
          <h2>
            Falcon 1
          </h2>
          <p>
            Falcon 1 description
          </p>
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rocket"
    >
      <img
        alt="Falcon 1"
        class="rocket__image"
      />
      <div
        class="rocket__info"
      >
        <h2>
          Falcon 1
        </h2>
        <p>
          Falcon 1 description
        </p>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket page Should render Cancel Reservation button when reserved = true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket page Should render Reserve Rocket button when reserved = true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket page renders rockets page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="rocket"
      >
        <div
          class="container"
        >
          <div
            class="rocket"
          >
            <img
              alt="Falcon 3"
              class="rocket__image"
            />
            <div
              class="rocket__info"
            >
              <h2>
                Falcon 3
              </h2>
              <p>
                Falcon description
              </p>
              <div />
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="rocket"
    >
      <div
        class="container"
      >
        <div
          class="rocket"
        >
          <img
            alt="Falcon 3"
            class="rocket__image"
          />
          <div
            class="rocket__info"
          >
            <h2>
              Falcon 3
            </h2>
            <p>
              Falcon description
            </p>
            <div />
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
